### DISCUSSION TITLE
The top-left icon of vedo window could be customized?

### DISCUSSION THREAD

**Discussion by ZiguoAtGitHub** [marcomusy/vedo#823] [Q&A] [answered]
Hi Marco,
I wander if the default icon at the top-left corner of plot window can be replaced by user's image and how to.
![image](https://user-images.githubusercontent.com/116484589/222099913-73b53651-2f7c-45e9-9809-017302a0b555.png)

Thanks!
:)

**Comment by marcomusy** [comment]
Hi, I think it's now well implemented in vtk in general... does this work on windows?
```python
from vedo import *
plt = Plotter()
s = Sphere()
pic = Picture("some.png")
plt.window.SetIcon(pic._data)
plt.show(s)
```

(will cause a crash on Linux machines, not tested on OSX).

**Reply by ZiguoAtGitHub** [reply]
Hi Marco,
I tried by the example script named icon.py and added the code "plt.window.SetIcon()", it doesn't work.😞
![image](https://user-images.githubusercontent.com/116484589/222140842-9a212167-f4f2-4d29-985c-3c2c4944c9dd.png)

**Reply by marcomusy** [reply]
you can still set an icon image with `Picture2D`, check out `examples/pyplot/embed_matplotlib.py`

**Reply by ZiguoAtGitHub** [reply]
Another new function!👍👍.
Looking forward to more! :)

**Reply by ZiguoAtGitHub** [reply]
Hi Marco,
I made a step forward.
When I add "plt.render()" before "plt.SetIcon(png._data)", the icon changed! However the icon is displayed as one black square like this.
![image](https://user-images.githubusercontent.com/116484589/222312760-abe7e39d-3991-4154-97b6-72b8bcbe1f60.png)
Could you give some advices  to me? 
Thanks a lot!
Here is my script and png file.
```
"""Make a icon to indicate orientation
and place it in one of the 4 corners
within the same renderer"""
from vedo import *

plt = Plotter(axes=5)

plt += Text3D(__doc__).bc('red')

elg = Picture('XVedo.png')

plt.add_icon(elg, pos=2, size=0.06)
plt.add_icon(VedoLogo(), pos=1, size=0.06)
plt.render()
plt.window.SetIcon(elg.inputdata())
plt.show().close()
```
![XVedo](https://user-images.githubusercontent.com/116484589/222313161-472f6e59-e79b-4564-ab1c-b39c75bc619d.png)

**Reply by marcomusy** [reply]
> Could you give some advices to me?

unfortunately i cannot reproduce the issue, i never work on windows systems..

**Reply by ZiguoAtGitHub** [reply, marked as answer]
😂
I will keep on trying.😃
